bundles/org.openhab.binding.hapero/src/main/java/org/openhab/binding/hapero/internal/handler/HaperoBridgeHandler.java

```diff
@@ -0,0 +1,629 @@
+                    faultOcurred = true;
+                    watchPath = null;
+                    errorMsg = "@text/offline.monitoring.failed";
+                }
+
+                if (watchPath != null) {
+                    try {
+                        watchService = FileSystems.getDefault().newWatchService();
+                        watchPath.register(watchService, StandardWatchEventKinds.ENTRY_MODIFY);
+                    } catch (Exception e) {
+                        logger.warn("Exception in Init: {}", e.getMessage());
+                        faultOcurred = true;
+                        errorMsg = "@text/offline.monitoring.failed";
+                    }
+                } else {
+                    logger.warn("Failed to create watchservice");
+                    faultOcurred = true;
+                    errorMsg = "@text/offline.monitoring.failed";
+                }
+
+                /* Start the File Watcher Runnable if a Watcher could be created */
+                if (!faultOcurred) {
+                    fileWatcherJob = scheduler.schedule(fileWatcherTask, 0, TimeUnit.SECONDS);
+                }
+            }
+
+            if (!faultOcurred) {
+                /*
+                 * If a watcher could be created, start a discovery.
+                 * The Thing status will be set to ONLINE after the first successful update
+                 */
+                updateStatus(ThingStatus.UNKNOWN);
+
+                scheduler.execute(() -> {
+                    updateThings();
+
+                    if (discoveryService != null) {
+                        discoveryService.startScan();
+                    }
+                });
+                logger.trace("Bridge initialized");
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, errorMsg);
+            }
+        } else {
+            /* A Watcher could not be created, set Thing to OFFLINE. Manual restart is required. */
+            cancelAllJobs();
+
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+            logger.trace("Bridge switched to offline mode because of configuration error");
+        }
+    }
+
+    /**
+     * Get a list of of all devices reported from the control system
+     *
+     * @return String List of all devices (may be empty)
+     */
+    public String[] getAllDevices() {
+        return deviceList.keySet().toArray(String[]::new);
+    }
+
+    /**
+     * Get the {@link Device} corresponding to the given deviceID
+     *
+     * @param deviceID ID for which to get the Device
+     * @return Device or null if the Device could not be found
+     */
+    public @Nullable Device getDevice(String deviceID) {
+        Device result = deviceList.get(deviceID);
+        if (result == null) {
+            logger.warn("getDevice: No device found for ID {}.", deviceID);
```

Comment:
I think this should remain as is because it notifies the user of an configuration problem. The heating system delivered data for an device that is not configured.